build/.azure-pipelines.TemplateValidation.yml

```
@@ -4,79 +4,209 @@ jobs:
+        net7_workloads: 'android ios maccatalyst'
+        net8_workloads: ''
+        install_net7: true
+        install_net8: false
+      MsalAuthMvvm_And_MauiBlank:
+        templateArgs1: '-preset=recommended -presentation mvvm -auth=msal'
+        templateArgs2: '-preset=blank -maui'
+        net7_workloads: 'android ios maccatalyst maui'
+        net8_workloads: ''
+        install_net7: true
+        install_net8: false
+      # Combining MauiRecommended and MauiRecommendedNet8 is causing build failures in CI:
+      # C:\hostedtoolcache\windows\dotnet\sdk\8.0.100-rc.1.23463.5\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(485,5): error NETSDK1112: The runtime pack for Microsoft.Maui.Core.Runtime.android was not downloaded. Try running a NuGet restore with the RuntimeIdentifier 'android'. [D:\a\1\s\UnoApp1\UnoApp1.Mobile\UnoApp1.Mobile.csproj::TargetFramework=net7.0-android]
+      # They are split for now. When adding a new validation, try to plug it in one of these two PLACEHOLDERs
       MauiRecommended:
-       templateArgs: '-preset=recommended -maui'
+        templateArgs1: '-preset=recommended -maui'
+        templateArgs2: 'PLACEHOLDER'
+        net7_workloads: 'android ios maccatalyst maui'
+        net8_workloads: ''
+        install_net7: true
+        install_net8: false
       MauiRecommendedNet8:
-       templateArgs: '-preset=recommended -maui -tfm net8.0'
+        templateArgs1: '-preset=recommended -maui -tfm net8.0'
+        templateArgs2: 'PLACEHOLDER'
+        net7_workloads: ''
+        net8_workloads: 'android ios maccatalyst maui'
+        install_net7: false
+        install_net8: true
 
   variables:
     - name: UseDotNetNativeToolchain
       value: false
 
   steps:
+  - checkout: none
+
+  - task: DownloadBuildArtifacts@0
+    inputs:
+      artifactName: $(Build.DefinitionName)
+
+  - script: |
+        md $(Build.SourcesDirectory)\src\PackageCache
+        copy "$(System.ArtifactsDirectory)\$(Build.DefinitionName)\*.nupkg" $(Build.SourcesDirectory)\src\PackageCache
+    displayName: Copy Artifacts to PackageCache
+
+  - script: dotnet new install "$(System.ArtifactsDirectory)\$(Build.DefinitionName)\Uno.Templates*.nupkg"
+    displayName: Install Project Templates
+
+  - powershell: |
+        dotnet nuget add source -n nuget_local $(Build.SourcesDirectory)\src\PackageCache
+        dotnet nuget add source -n uno_dev "https://pkgs.dev.azure.com/uno-platform/1dd81cbd-cb35-41de-a570-b0df3571a196/_packaging/unoplatformdev/nuget/v3/index.json"
+        dotnet nuget add source -n net8 "https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet8/nuget/v3/index.json"
+    # TODO: Maybe we shouldn't add these feeds for non-canary builds.
+    displayName: Add NuGet feeds
+
+  - task: UseDotNet@2
+    displayName: 'Use .NET 7 SDK'
+    inputs:
+      version: '7.0.400'
+    condition: and(succeeded(), eq(variables['install_net7'], 'true'))
```

```suggestion
    condition: and(succeeded(), eq(${install_net7}, 'true'))
```
I think this should work

---

Will try it, but IIRC, I tried something similar to it yesterday and didn't work. I found that the `variable['VAR_NAME']` syntax is extensively used by Uno CI so went for it.